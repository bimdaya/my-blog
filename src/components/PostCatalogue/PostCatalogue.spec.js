import React from 'react';
import { render } from '@testing-library/react';
import PostList from './PostCatalogue';

describe('PostList', () => {
  it('renders without passing posts', () => {
    const { container } = render(<PostList />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders when an array of posts are passed', () => {
    const posts = [
      {
        slug: 'test_slug',
        title: 'test title',
        date: '2020.04.05',
        description: 'test description',
      },
    ];
    const { container } = render(<PostList posts={posts} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
