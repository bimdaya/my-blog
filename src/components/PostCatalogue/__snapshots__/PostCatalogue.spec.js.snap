// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList renders when an array of posts are passed 1`] = `
<section>
  <h2>
    Grab a cuppa ☕️ and enjoy the reading 📖
  </h2>
  <div>
    <div
      class="post_margin"
    >
      <h4>
        <a
          class="post_name"
          href="/posts/test_slug"
        >
          📋 
          test title
          ,
          <span
            class="post_published_date"
          >
             📆 
            2020.04.05
          </span>
        </a>
      </h4>
      <p
        class="post_description"
      >
        test description
      </p>
    </div>
  </div>
</section>
`;

exports[`PostList renders without passing posts 1`] = `
<section>
  <h2>
    Grab a cuppa ☕️ and enjoy the reading 📖
  </h2>
  <div />
</section>
`;
